kernel/src/common/utils.rs

```diff
@@ -22,14 +22,27 @@ macro_rules! static_init {
             // initial value into it (without dropping the initial zeros) and
             // return a reference to it.
             static mut BUF: MaybeUninit<$T> = MaybeUninit::uninit();
-            BUF.as_mut_ptr().write($e);
-            // TODO: use MaybeUninit::get_mut() once that is stabilized (see
-            // https://github.com/rust-lang/rust/issues/63568).
-            &mut *BUF.as_mut_ptr() as &'static mut $T
+            $crate::static_init_half!(&mut BUF, $T, $e)
         };
     }
 }
 
+/// Same as `static_init!()` but without actually creating the static buffer.
+/// The static buffer must be passed in.
+#[macro_export]
+macro_rules! static_init_half {
```

Comment:
I am not convinced this needs to be a macro at all as
```
pub unsafe fn static_init_half<T>(buf: &'static mut MaybeUninit<T>, val: T) -> &'static mut T {
    buf.as_mut_ptr().write(val);
    // TODO: use MaybeUninit::get_mut() once that is stabilized (see
    // https://github.com/rust-lang/rust/issues/63568).
    &mut *buf.as_mut_ptr()
}
```
should work just as well, but I guess better to do one step at a time..